RELEASE.md

```diff
@@ -0,0 +1,138 @@
+
+For this:
+1. Make sure you have downloaded the `$BUILDER_TAG` builder's binary locally `slsa-builder-go-linux-amd64`, either via the web UI or via:
+```
+$ "$GH" release -R slsa-framework/slsa-github-generator download "$BUILDER_TAG" -p "slsa-builder-go-linux-amd64"
+$ mv slsa-builder-go-linux-amd64 slsa-builder-go-linux-amd64-"$BUILDER_TAG".original
+```
+2. Upload a different binary to the assets:
+```
+$ echo hello > slsa-builder-go-linux-amd64
+$ "$GH" release -R slsa-framework/slsa-github-generator upload "$BUILDER_TAG" slsa-builder-go-linux-amd64  --clobber
+```
+3. Update the version of the workflow [slsa-framework/example-package/.github/workflows/e2e.go.workflow_dispatch.main.adversarial-builder-binary.slsa3.yml#L14](https://github.com/slsa-framework/example-package/blob/main/.github/workflows/e2e.go.workflow_dispatch.main.adversarial-builder-binary.slsa3.yml#L14) with the `$BUILDER_TAG` to test.
+4. Trigger the test in [slsa-framework/example-package/actions/workflows/e2e.go.workflow_dispatch.main.adversarial-builder-binary.slsa3.yml](https://github.com/slsa-framework/example-package/actions/workflows/e2e.go.workflow_dispatch.main.adversarial-builder-binary.slsa3.yml) by cliking `Run workflow`. Verify that it fails, with a message:
+```
+verifier hash computed is 60c91c9d5b9a059e37ac46da316f20c81da335b5d00e1f74d03dd50f819694bd
+verifier hash verification has passed
+verification failed: could not find a matching signature entry
+Error: Process completed with exit code 6.
+```
+
+If it does not, delete the release, fix the bug and re-start the release process at the top of this page.
+
+4. If the test above failed with the expected message, re-upload the original binary back to the assets, e.g. via:
+```
+$ mv slsa-builder-go-linux-amd64-"$BUILDER_TAG".original slsa-builder-go-linux-amd64
+$ "$GH" release -R slsa-framework/slsa-github-generator upload "$BUILDER_TAG" slsa-builder-go-linux-amd64  --clobber
+```
+
+5. Re-run the workflow above and verify that it succeeds. (TODO: we need a set of workflows identical to the e2e tests but using pre-built binaries; and do this as first pre-release tests)
+
+## Update verifier
+
+The next step is to update the verifier's e2e tests. For this, you need to:
+
+1. Generate binaries and provenance for a project, using the `$BUILDER_REF` builder. TODO: need a common repository to do this.
+
+2. Place the files in a new directory `slsa-framework/slsa-verifier/tree/main/testdata/$BUILDER_TAG`.
+
+3. Add the new release to the list defined in [slsa-framework/slsa-verifier/blob/main/main_test.go](https://github.com/slsa-framework/slsa-verifier/blob/main/main_test.go).
+
+Send a pull request to merge the changes into the verifier's repository. The pre-submits will validate that the verifier is able to verify provenance from the `$BUILDER_TAG` builder.
+
+## Finalize release
+
+Untick the `This is a pre-release` option. That's it!
+
+## Update the starter workflow
+
+TODO
```

Comment:
You can probably comment these out as todos and put in an issue to update them.

Reply:
Created the issues.